dev/com.ibm.ws.repository/src/com/ibm/ws/repository/resources/internal/AppliesToProcessor.java

```diff
@@ -247,19 +245,34 @@ public static void validateEditions(AppliesToFilterInfo info, String appliesToHe
     }
 
     /**
-     * Similiar logic to parseAppliesToHeader above, duplicate logic, be nice to find time to factor this into
-     * a common routine
-     *
-     * @param appliesTo
-     * @return
+     * Parse the given raw appliesTo header into a list
+     * of AppliesToEntry objects
      */
     public static List<AppliesToEntry> parseAppliesToEntries(String appliesTo) {
         List<AppliesToEntry> result = new ArrayList<AppliesToEntry>();
+        List<List<String>> parsedProducts = doParse(appliesTo);
+        for (List<String> parsedProduct : parsedProducts) {
+            AppliesToEntry entry = new AppliesToEntry();
+            for (String thing : parsedProduct) {
+                entry.add(thing);
```

Comment:
`thing` isn't a great name

`productChunk` as used above might be better.

Also, it's not obvious reading the code what the difference between an `AppliesToEntry` and an `AppliesToFilterInfo` is. I assume we need methods for both though?